datafusion/core/src/dataframe/parquet.rs

```diff
@@ -166,4 +166,47 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn write_parquet_with_small_rg_size() -> Result<()> {
+        let mut test_df = test_util::test_table().await?;
+        // make the test data larger so there are multiple batches
+        for _ in 0..7 {
+            test_df = test_df.clone().union(test_df)?;
+        }
+        let output_path = "file://local/test.parquet";
+
+        for rg_size in (1..7).step_by(5) {
```

Comment:
My reading of the [docs](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.step_by) and my [playground experiments](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2ff81b06e8447214338607095c726d7c) suggests this is the same as `[1, 6]` -- is that the intent? Or did you mean `1, 5, 10, 15, 20, 25, 30, 35`?

Reply:
Yes, [1, 6] is all I meant and would be clearer... I originally wanted to loop over more rg sizes but the test was slow. If we streamline the test, we could actually range over more values here.

Reply:
This now loops over 0..10 with datafusion.execution.batch_size set to 10.